Plugins/Public/base_plugin/PlayerBase.cpp

```diff
@@ -366,33 +367,49 @@ void PlayerBase::Load()
 			{
 				CoreModule* mod = new CoreModule(this);
 				mod->LoadState(ini);
-				modules.emplace_back(mod);
+				modules.at(moduleCounter) = mod;
```

Comment:
vector::at() returns the reference to the element at a given index.
I've had to change it up for the purpose of being able to set up modules in an order different than counting from zero, specifically to be able to put a core upgrade into a separate core-upgrade-only slot at the very end. Which meant I've had to resize the vector ahead of time, causing me to insert modules to pre-existing slots instead of pushing/emplacing them to the back.

Reply:
I get that but assigning the result of a function call?

Reply:
Again, it returns a _reference_ to the element, so we can assign to it safely.